src/engine/source/api/src/test/sessionManager.cpp

```diff
@@ -0,0 +1,124 @@
+#include "api/test/sessionManager.hpp"
+
+#include <logging/logging.hpp>
+
+namespace api::sessionManager
+{
+
+SessionManager& SessionManager::getInstance(void)
+{
+    static std::once_flag flag;
+    static std::unique_ptr<SessionManager> instance;
+
+    // Thread-safe initialization.
+    std::call_once(flag, []() { instance.reset(new SessionManager); });
+
+    return *instance;
+}
+
+std::optional<base::Error>
+SessionManager::createSession(const string& sessionName, const string& policyName, uint32_t lifespan)
+{
+    std::lock_guard<std::mutex> lock(m_sessionMutex);
+
+    if (m_activeSessions.count(sessionName) > 0)
+    {
+        return base::Error {fmt::format("Session name '{}' already exists", sessionName)};
+    }
+
+    if (m_policyMap.count(policyName) > 0)
+    {
+        return base::Error {
+            fmt::format("Policy '{}' is already assigned to a route ('')", policyName, m_policyMap[policyName])};
+    }
+
+    // TODO: interact with router to get a valid route
+    // routeName = getRoute(policyName);
+    auto routeName = "routeName";
+
+    Session session(sessionName, policyName, routeName, lifespan);
+    m_activeSessions.emplace(sessionName, session);
+    m_routeMap[routeName] = sessionName;
```

Comment:
improve-desing: Why isn't the emplace() method still used? with this you avoid creating explicit temporary objects